README.md

```diff
@@ -6,27 +6,392 @@
+
+> Installation instructions for dependencies will vary based on your OS and can
+> change over time, so please refer to the official documentation for each dependency
+> on how to set them up correctly.
+
+Now, follow the steps below in the `backend` directory.
+
+1. Copy `.env.example` to `.env`.
+
+```shell
+cp .env.example .env
+```
+
+> This file contains environment variables used by the project. You can modify these
+> to suit your local development environment.
+
+2. Copy `local_settings_example.py` to `local_settings.py`.
+
+```shell
+cp local_settings_example.py local_settings.py
+```
+
+> These can be used to modify settings for local development without changing the main settings file.
+
+3. Create a virtual environment & install dependencies.
+
+```shell
+poetry install
+```
+
+4. Add pre-commit hooks
+
+```shell
+poetry run pre-commit install
+```
+
+5. Run migrations
+
+```shell
+poetry run python manage.py migrate
+```
+
+6. Generate test data
+
+```shell
+poetry run python manage.py create_test_data
+```
+
+7. Start the server
+
+```shell
+poetry run manage.py runserver localhost:8000
+```
+
+Backend should now be running at `localhost:8000`.
+
+> Since the backend is not in the root of the project, the source path is not correct for linting out of the box.
+> Fixing this is specific to your IDE, but here are some setups for popular ones:
+>
+> PyCharm: Right click on the `backend` folder and select "Mark Directory as" and then select "Sources Root".
+>
+> VSCode: In your workspace settings (`.vscode/settings.json`), add the following:
+>
+> ```json
+> {
+>     "python.analysis.extraPaths": [
+>         "$(workspaceFolder)/backend"
+>     ]
+> }
+> ```
+
+### Linting
+
+It's recommended to set up [Ruff] linting and formatting support in your editor.
+
+[Ruff]: https://docs.astral.sh/ruff/editors/setup/
+
+### Testing
+
+Tests are run with `pytest`.
+
+Some flags that can save time when running tests:
+
+- To skip slow-running tests: `pytest --skip-slow`
+- To retain test database between runs: `pytest --reuse-db`
+- To skip migration-checks at the start of tests: `pytest --no-migrations`
+- To run tests in parallel: `pytest -n 8 --dist=loadscope` (=8 cores, use `-n auto` to use all available cores)
+
+You can use a `pytest.ini` file to set up flags for local development.
+
+### Updating dependencies
+
+Dependencies are managed by [Poetry]. Normally, they are automatically updated by [dependabot]
+without any manual intervention (given updates don't fail any automated tests).
+
+[Poetry]: https://python-poetry.org/
+[dependabot]: https://github.com/dependabot
+
+However, if you want to update them manually, you can do so by running:
+
+```shell
+poetry update
+```
+
+This will update all dependencies according to the rules defined in `pyproject.toml`.
+To see all outdated dependencies, run:
+
+```shell
+poetry show -o
+```
+
+Note that this will also include any sub-dependencies that are not directly defined in `pyproject.toml`.
```

Comment:
Might be good to clarify here that since we define specific versions in `pyprojecy.toml` running `poetry update` does not really update our direct dependencies, only sub-dependencies

To update our dependencies defined in `pyprojecy.toml`, one should use `poetry add ...` or manually update the file and run `poetry lock && poetry install`

However Dependabot will update `pyprojecy.toml` dependency versions

Reply:
Updated the dependency update instructions.